scripts/thaliPrerequisite.js

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env node
+/*
+ This script will do the following tasks
+ 0. Clone the working jxcore-cordova plugin
+ 1. Fix issue on "can not replace existing file"
+ 2. Install the thali node module to global (until we get thali in npm repo)
+ 3. Move the build-extras.gradle to platform
+ 4: Create a dummy file to prevent future execution by other plugin add operation
+ */
+
+
+var fs = require('fs');
+var exec = require('child_process').execSync;
+
+
+var rootdir = process.argv[2];
+//Dummy file to prevent the execution of this script when the developer add other Cordova plugins
+var dummyFile = rootdir + '/../plugins/org.thaliproject.p2p/dummy.js';
+
+//Replace JXcoreExtension.java
+//Copy from \plugins\org.thaliproject.p2p\src\android\java\io\jxcore\node to \platforms\android\src\io\jxcore\node
+function replaceJXCoreExtension(){
+    var sourceFile =  rootdir + '/../plugins/org.thaliproject.p2p/src/android/java/io/jxcore/node/JXcoreExtension.java';
+    var targetFile = rootdir + '/../platforms/android/src/io/jxcore/node/JXcoreExtension.java';
+    fs.writeFileSync(targetFile, fs.readFileSync(sourceFile));
+}
+
+//Check whether this is the first time this script is getting executed or not
+//When ever this script run, it creates the dummyfile at the end of execution
+function isFirstTime(){
+    if (fs.existsSync(dummyFile))
+        return false;
+    else
+        return true;
+}
+
+
+//Install thali node modules to global
+//once we have the thali module available in npm repo, this is step is not required
+function installThaliModules(){
+    var thaliPath = rootdir + '/../plugins/org.thaliproject.p2p/thali';
+    var currentFolder = process.cwd();
+    process.chdir(thaliPath);
+    //exec('jx install -g --autoremove "*.gz"'); // https://github.com/jxcore/jxcore/issues/429
+    exec('jx install -g');
+    process.chdir(currentFolder);
+}
+
+
+//TODO: This is a temporary fix, and we don't require this method once we get the JXCore fix in master
+//Uncomment the dependency section in plugin.xml while removing this method
+function updateJXCore(){
+    process.chdir(process.cwd());
+    console.log('Cloning the jxcore-cordova repo (380MB size)..it might take a while. Please be patient..');
+    exec('git clone -b 0.0.3-dev --single-branch https://github.com/jxcore/jxcore-cordova.git');
```

Comment:
I think that using chdir to navigate before issuing commands isn't a great idea. I believe it makes the code hard to read (since the setup for the command is across two lines) and it is error prone (you need to remember to navigate back). Instead I would suggest getting rid of chdir and when using exec both here and in line 45 (and 58 and so on) just using the options argument to submit a CWD value.

I would also suggest that all exec's should have a timeout value in their options as well to make sure that the system doesn't just hang.